scpdt/plugin.py

```diff
@@ -3,21 +3,103 @@
 """
 
 from _pytest import doctest
+from _pytest.doctest import DoctestModule, DoctestTextfile
+from _pytest.pathlib import import_path
+from _pytest.outcomes import skip
 
-from scpdt.impl import DTChecker, DTConfig
+from scpdt.impl import DTChecker, DTConfig, DTParser, DTFinder
 
 
 def pytest_configure(config):
     """
     Allow plugins and conftest files to perform initial configuration.
     """
-    if config.pluginmanager.getplugin('doctest'):
-        config.pluginmanager.register(DTChecker())
 
     doctest._get_checker = _get_checker
+    doctest.DoctestModule = DTModule
+    doctest.DoctestTextfile = DTTextfile
+
 
 def _get_checker():
     """
     Override function to return an instance of DTChecker with default configurations
     """
     return DTChecker(config=DTConfig())
+
+
+class DTModule(DoctestModule):
+    """
+    Override class to use an instance of DTFinder which utilizes the DTParser to filter out stopwords
+    and pseudocode when creating a DoctestItem
+    """
+    def collect(self):
+        if self.path.name == "setup.py":
+            return
+        if self.path.name == "conftest.py":
+            module = self.config.pluginmanager._importconftest(
+                self.path,
+                self.config.getoption("importmode"),
+                rootpath=self.config.rootpath
+            )
+        else:
+            try:
+                module = import_path(
+                    self.path,
+                    root=self.config.rootpath,
+                    mode=self.config.getoption("importmode"),
+                )
+            except ImportError:
+                if self.config.getvalue("doctest_ignore_import_errors"):
+                    skip("unable to import module %r" % self.path)
+                else:
+                    raise
+        finder = DTFinder(config=DTConfig())
+        optionflags = doctest.get_optionflags(self)
+        runner = doctest._get_runner(
+            verbose=False,
+            optionflags=optionflags,
+            checker=_get_checker(),
+            continue_on_failure=doctest._get_continue_on_failure(self.config),
+        )
+
+        for test in finder.find(module, module.__name__):
+            if test.examples:  # skip empty doctests
+                yield doctest.DoctestItem.from_parent(
+                    self, name=test.name, runner=runner, dtest=test
+                )
```

Comment:
This whole class looks scary. Where does all this code come from? If it's the pytest implementation, copy-pasted and tweaked, please say just this in the docstring/code comment, add a link to the original (incl the version of pytest this is taken from) and mention the changes. For the latter maybe even add a diff in the commit message. 

Also, please spell out the rationale: is it that you had to subclass DoctestModule just to add `_get_parser`? if yes, please spell this out in a comment, too.